gateway/activation/errors.go

```diff
@@ -0,0 +1,46 @@
+package activation
+
+import (
+	"fmt"
+
+	"github.com/spacemeshos/poet/shared"
+)
+
+var (
+	ErrAtxNotFound = &AtxNotFoundError{}
+	ErrTransport   = &TransportError{}
+)
+
+// AtxNotFoundError means that ATX with given ID was not found.
+type AtxNotFoundError struct {
+	id shared.ATXID
+}
+
+func (e *AtxNotFoundError) Error() string {
+	return fmt.Sprintf("ATX %X not found", e.id)
+}
+
+func (e *AtxNotFoundError) Is(target error) bool {
+	_, ok := target.(*AtxNotFoundError)
+	return ok
+}
+
+// TransportError means there was a problem communicating
+// with Activation Service.
+type TransportError struct {
+	// additional contextual information
+	msg string
+	// the source (if any) that caused the error
+	source error
+}
+
+func (e *TransportError) Error() string {
+	return fmt.Sprintf("transport error: %s (%v)", e.msg, e.source)
+}
+
+func (e *TransportError) Unwrap() error { return e.source }
+
+func (e *TransportError) Is(target error) bool {
```

Comment:
Same as above, this checks if one `TransportError` is the same as any other `TransportError`, not if they are actually equal, for this `msg` and `source` should be checked for equality.

If just the type is relevant for the check, `errors.As` is imo the better check.

Reply:
Removed `Is()` implementation as not required.